test/i18n_viz_test.rb

```diff
@@ -51,3 +51,36 @@ class I18nVizJavascriptIntegrationTest < ActionDispatch::IntegrationTest
     assert page.has_css?("#i18n_viz_tooltip")
   end
 end
+
+class I18nVizDynamicUrlTest < ActionDispatch::IntegrationTest
+  setup do
+    Capybara.current_driver = Capybara.javascript_driver # :selenium by default
+
+    # monkey patch external_tool_url
+    I18nViz::Middleware.class_eval do
+      alias orig_external_tool_url external_tool_url
+      def external_tool_url
+        lambda { |x| "http://example.com/?s=" }
```

Comment:
shouldn't the tool URL have some dynamic part, e.g. the current locale `I18n.locale`?